src/panel/direction/MobileRoadMapPreview.jsx

```diff
@@ -16,14 +16,25 @@ const MobileRoadMapPreview = ({
   const scroll = () => {
 
     const newStep = Math.floor(
-      // Divide the step container's scrollLeft up to the middle of the screen with the size of a step
-      // to determine which step is present at the middle of the screen
-      (stepsRef.current.scrollLeft + window.innerWidth / 2) / (window.innerWidth - 70 + 12)
+      // Determine which step is fully visible on screen
+      stepsRef.current.scrollLeft / (window.innerWidth - 70 + 12)
     );
 
-    // If it has changed, save it and highlight it as the current step
+    // If it is different from the current step:
     if (currentStep !== newStep) {
+
+      // Save it and highlight it as the current step
       setCurrentStep(newStep);
+
+      // Stop scroll inertia (except when going to previous step, which happens at the start of the scroll)
+      if ((currentStep - 1) !== newStep) {
+        stepsRef.current.style.overflow = 'hidden';
+        stepsRef.current.scrollLeft = stepsRef.current.scrollLeft;
```

Comment:
This is really strange as it's just a self assign, but when I remove it I have a kind of glitch, like a jump at the end of the transition, so it's probably useful in a way but we have to understand it better.
What was the idea here?